### PR TITLE
lazy json so we can store all the json elsewhere but still interface

### PR DESCRIPTION
**Pull request by CJ-Wright**
with it like normal

### CODE REVIEW

File: conda_forge_tick/utils.py [hunk 1]
```diff
@@ -42,6 +42,56 @@ def __getitem__(self, name):
...
+                self.data = json.load(f)
+
+    def _dump(self):
+        self._load()
+        with open(self.file_name, "w") as f:
+            json.dump(self.data, f)
+
+    def __getitem__(self, item):
+        self._load()
+        return self.data.__getitem__(item)
```
Comment (scopatz) [inline conda_forge_tick/utils.py]: This should be `return self.data[item]`

File: conda_forge_tick/utils.py [hunk 2]
```diff
@@ -42,6 +42,56 @@ def __getitem__(self, name):
...
+
+    def __setitem__(self, key, value):
+        self._load()
+        self.data.__setitem__(key, value)
```
Comment (scopatz) [inline conda_forge_tick/utils.py]: Should be `self.data[key] = value`

**Comment by CJ-Wright** [comment]
ping @scopatz

File: conda_forge_tick/utils.py [hunk 3]
```diff
@@ -42,6 +42,56 @@ def __getitem__(self, name):
...
+                json.dump({}, f)
+        self.data = None
+
+    def __len__(self) -> int:
+        self._load()
+        return len(self.data)
+
+    def __iter__(self):
+        self._load()
+        return self.data.__iter__
```
Comment (scopatz) [inline conda_forge_tick/utils.py]: Shouldn't this be a `yield from self.data`?

**Comment by CJ-Wright** [comment]
one more round? @scopatz

File: conda_forge_tick/utils.py [hunk 4]
```diff
@@ -42,6 +42,56 @@ def __getitem__(self, name):
...
+                json.dump({}, f)
+        self.data = None
+
+    def __len__(self) -> int:
+        self._load()
+        return len(self.data)
+
+    def __iter__(self):
+        self._load()
+        yield from self.data.__iter__
```
Comment (scopatz) [inline conda_forge_tick/utils.py]: This should just be `yield from self.data`

**Comment by CJ-Wright** [comment]
Thank you!

**Pull request merged**